omod/src/main/java/org/openmrs/module/spa/SpaActivator.java

```diff
@@ -13,17 +13,22 @@
 import org.openmrs.api.GlobalPropertyListener;
 import org.openmrs.api.context.Context;
 import org.openmrs.module.BaseModuleActivator;
+import org.springframework.web.context.ServletContextAware;
+
+import javax.servlet.ServletContext;
 
 @Slf4j
-public class SpaActivator extends BaseModuleActivator {
+public class SpaActivator extends BaseModuleActivator implements ServletContextAware {
```

Comment:
I don't think we can meaningfully create a bean out of an Activator. It's handled by core, not Spring. Besides which, we should be able to get a servletContext at most points we wound need it.

Reply:
You can probably get a servletContext but abusing the `init()` method on the filter and storing it in a static variable.

Reply:
This is how the same method OWA module handles this: https://github.com/openmrs/openmrs-module-owa/blob/3d0f1f07be247fe0fbbc4b9713005a57f10cfe93/omod/src/main/java/org/openmrs/module/owa/activator/OwaActivator.java#L35

Reply:
Yeah, now that you point that out, I did basically the same thing in the FHIR2 module. It's ugly, but I guess it works...